### PR TITLE
Implemented Chat Completions endpoint

### PR DESCRIPTION
**Pull request by dehidehidehi**

### CODE REVIEW

**Comment by hntan** [comment]
Openai API is quite hot right now but looks like this repo maintainers are not quite active recently

**Comment by jiangying000** [comment]
Is there an easy way that I can use the package with this version of code ?
Like change version in pom.xml to 0.10.1.

File: api/src/main/java/com/theokanning/openai/completion/chat/ChatCompletionResult.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * A list of all generated completions.
+     */
+    List<ChatCompletionChoice> choices;
+
+    /**
+     * The API usage for this request.
+     */
+    Usage usage;
+
+}
```
Comment (jiangying000) [inline api/src/main/java/com/theokanning/openai/completion/chat/ChatCompletionResult.java:38]: did you missed model param here ?
this is an sample response
```json
{
  "id": "chatcmpl-6peKOXv1MPTwR727Ib70CfcPYjrSc",
  "object": "chat.completion",
  "created": 1677767768,
  "model": "gpt-3.5-turbo-0301",
  "usage": {
    "prompt_tokens": 44,
    "completion_tokens": 35,
    "total_tokens": 79
  },
  "choices": [
    {
      "message": {
        "role": "assistant",
        "content": "根据维基百科，刘若英是1968年6月12日出生的，目前是53岁。"
      },
      "finish_reason": "stop",
      "index": 0
    }
  ]
}
```

Comment (dehidehidehi) [reply api/src/main/java/com/theokanning/openai/completion/chat/ChatCompletionResult.java:38]: Good catch, I'll verify this when I get home :)

**Comment by TheoKanning** [comment]
Tested this locally and it all worked, thanks so much for doing this! I couldn't get early access to the chat-got api, and of course they had to release it the week I start a new job haha

**Pull request merged**